tests/blob-store/combine-states.js

```diff
@@ -97,31 +97,44 @@ test('arithmetic test', (t) => {
 
 /**
  * Returns an iterator of all permutations of the given array.
- * From https://stackoverflow.com/a/37580979/3071863
+ *
+ * Implements [Heap's algorithm][0].
+ *
+ * [0]: https://en.wikipedia.org/wiki/Heap%27s_algorithm
+ *
  * @template T
- * @param {Array<T>} arr
- * @returns {IterableIterator<Array<T>>}
+ * @param {ReadonlyArray<T>} arr
+ * @returns {IterableIterator<ReadonlyArray<T>>}
  */
-export function* permute(arr) {
-  var length = arr.length,
-    c = Array(length).fill(0),
-    i = 1,
-    k,
-    p
```

Comment:
I had too much trouble fixing this so I just reimplemented it (using [a solution I wrote last year][0]).

[0]: https://evanhahn.com/nested-list-permutations-in-javascript/#flat-permutations